core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -2070,8 +2070,8 @@ protected Type visitParameter(Parameter node, StackableAstVisitorContext<Context
             if (parameters.size() == 0) {
                 throw semanticException(INVALID_PARAMETER_USAGE, node, "Query takes no parameters");
             }
-            if (node.getPosition() >= parameters.size()) {
-                throw semanticException(INVALID_PARAMETER_USAGE, node, "Invalid parameter index %s, max value is %s", node.getPosition(), parameters.size() - 1);
+            if (node.getId() >= parameters.size()) {
```

Comment:
But if it is some sort of a uniqueId (which I totally agree) then comparing `getId` but `parameters` size would make less sense.

Reply:
That's sort of true, unless we define the ids as being from 0 to the number of parameters in the query. I hesitated removing these checks, but I do agree they look wonky. Let me see if there's a better way to represent them, or I'll just remove them.